be/src/runtime/types.h

```diff
@@ -225,7 +225,7 @@ struct TypeDescriptor {
 
     bool is_implicit_castable(const TypeDescriptor& from) const {
         if (is_decimal_type()) {
-            return precision == from.precision && scale == from.scale;
+            return true;
```

Comment:
I suggest to remove this function and just put this implementation at orc level.

Because that castable only works in orc scenario.